CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+    ```
+
+    Refer to [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) for more details.
+
+* Install [pre-commit](https://pre-commit.com/#install) to automatically ensure that your commits comply with our code style for both C++ and JavaScript. Note this is currently not working on Windows. This saves time reviewing so we don't have to point out nitpicky style issues. Once you have pre-commit installed on your computer, set it up in your local Git repository:
+
+    ```
+    cd /path/to/your/git/repo
+    pre-commit install
+    pre-commit install -t pre-push
+    ```
+
+    If you have a problems with a particular hook, you can use the SKIP environment variable to disable hooks:
+
+    ```
+    SKIP=clang-format,end-of-file-fixer git commit
+    ```
+
+    This can also be used to separate logic changes and autoformatting into two subsequent commits. Using the SKIP environment variable is preferable to using `git commit --no-verify` (which also disables the checks) because it won't prevent catching other, unrelated issues.
+
+* Generally, prefer merging to rebasing. Do not rebase unless you have discussed that with whoever is reviewing the pull request. When you rebase a branch with an open pull request, prior review comments made inline in the code on GitHub lose their connection to that spot in the code. If you want to correct minor mistakes with a rebase or `git commit --amend` within a few minutes of pushing commits, that is okay as long as no one has started reviewing those commits yet.
+* If you are helping with someone else's pull request that is not yet merged, open a pull request targeted at their fork. Leave a comment on the upstream pull request (which targets mixxxdj/mixxx) with a link to your pull request so other Mixxx contributors are aware of your changes.
+* Low risk bug fixes should be targeted at the stable branch (currently 2.2). However, bug fixes for the stable branches must have a direct impact on users. If you spot a minor bug reading the code or only want to clean up the code, target that at the master or beta branch.
+* Controller mappings should be targeted at the stable branch unless they use features that are new in the beta or master branch.
+* If you are making changes to the GUI with a pull request, please post before and after screenshots of the changes.
+* Please help review other people's pull requests. When others review your pull requests, please return the favor. The continued progress of Mixxx depends on all of us working together. Even if you are not familiar with the area of the code being changed in a pull request, you can be helpful by building the branch, verifying that it works as described, and commenting with feedback about the user experience design.
+* If you demonstrate good coding skills, help review pull requests, contribute major features, and show a commitment to Mixxx over time, we may invite you to the core team.
```

Comment:
```suggestion

If you demonstrate good coding skills, help review pull requests, contribute major features, and show a commitment to Mixxx over time, we may invite you to the core team.
```

I think this should be more of a conclusory/bridging paragraph than an extra point